archivers/brotli/files/patch-disable-cmake-ver-check-CmakeLists.txt.diff

```diff
@@ -0,0 +1,12 @@
+--- CMakeLists.txt	2023-12-14 01:10:33
++++ CMakeLists.txt	2023-12-14 01:10:58
+@@ -1,7 +1,8 @@
+ # Available CMake versions:
+ #  - Ubuntu  20.04 LTS   : 3.16.3
+ #  - Solaris 11.4 SRU 15 : 3.15
+-cmake_minimum_required(VERSION 3.15)
++# Disable minimum CMake version check to allow use of cmake-bootstrap
++# cmake_minimum_required(VERSION 3.15)
```

Comment:
Yes, we do. But we're not using it :)

```
# use cmake-bootstrap to avoid circular dependency:
# cmake depends on curl which depends on brotli.
depends_build-replace \
                    path:bin/cmake:cmake port:cmake-bootstrap
configure.cmd       ${prefix}/libexec/cmake-bootstrap/bin/cmake
```

`brotli` builds using `cmake-bootstrap`, which is pegged at v3.9.6.

Reply:
> disabling the cmake minimum set by the build may work for now to allow cmake-bootstrap, but you would imagine that it will be very fragile for the future.

I agree. What do you think we should do instead?

I imagine that if we ever do get to the point where `brotli` _actually_ does require the CMake features from the version it's normally checking for, the build should fail if we force the use of CMake versions below that.

Reply:
yeah - ugh. what do we have as options here...

1. do this, and hope to kick the can down the road a few years
2. "brotli-bootstrap" that builds with cmake-bootstrap that curl can use. yuk
3. change curl to not use brotli, if that is possible

that it? well if that is it, option 1 looks easiest for the foreseeable future until one of the other two is needed.

Reply:
It sounds like it's going to be option 1 for now.

@raimue, thoughts?

CC: @catap

Reply:
Well, going via (1) seems like an easy solution. If we decoded this one, I suggest to open an issue upstream to raise concern about this issue.

Anyway, I see another way which seems like the best move: move support of curl at cmake into variant.

curl is used at cmake to fetch artifacts on build stage. It still can be avoided by spending a bit more, and mine `retdec` port has everything which can be moved to cmake PG to make it easy to use ;)

Reply:
Thanks for the answer, that explains it. I knew I must have missed something.

Yes, I agree (1) is the most viable option for now. Let's not anticipate too much.

I also checked what FreeBSD ports is doing for this [brotli update](https://cgit.freebsd.org/ports/diff/archivers/brotli?id=62740dc1077c3c403e74ec6cae3a8b437ddb57d6), but their [cmake](https://cgit.freebsd.org/ports/tree/devel/cmake-core/Makefile?id=6c4806b462e98ff76c087ca9ca81dfc96136061d) is built without libcurl.